### DISCUSSION TITLE
teensy4.1+T41U5XBB wifi

### DISCUSSION THREAD

**Discussion by distebia** [phil-barrett/grblHAL-teensy-4.x#104] [General]
Salve, esiste qualche possibilità per avere il teensy4.1+T41U5XBB con connessione wifi? se si come fare?

**Comment by phil-barrett** [comment]
Sorry, not possible with the T41U5XBB.  The [PicoCNC,](https://www.grbl.org/picocnc-1) on the other hand, can use the Pico W which has WiFi.

**Comment by distebia** [comment]
Salve, non è un problema ma ho una domanda che mi gira da tanto tempo in testa....il teensy4.1 è equipaggiato con ARM Cortex-M7 a 600 MHz...posso in qualche modo paragonarlo ai controller ruida? Non mi riferisco ne al firmware (con tutte le opzioni disponibili) ne a tutte le protezioni, accorgimenti del controller ecc... che sono presenti nel ruida...Parlo solo di velocità di calcolo e capacità di avere ottime velocità di raster (>50000mm/min) anche con una struttura corexy...è un confronto valido?

**Reply by phil-barrett** [reply]
Sorry, my Italian is not very good so responding in English.

The Ruida controller is DSP based (they do not say which DSP but I believe it is an Altera Cyclone II).  The Teensy 4.1 uses an iMXRT1062.  It is very fast and grblHAL supports Lightburn clusters for even better speed.  You are correct - the Teensy 4.1 can perform very high speed laser operations.  When configured with a large planning buffer ($398=1024), it handles short segments quite well.  I have a number of customers using the Teensy 4.1 for laser CNC machines doing production work. With the right mechanism (like core XY), grblHAL on a Teensy can drive well faster than 50K mm/min. It can generate step rates of up to 400 kHz with very little jitter.

**Reply by distebia** [reply]
ne vorrei costruire una con cinematica core xy, teensy 4.1 su t41u5xbb, driver esterni tb6600 oppure dm556, non so se usare nema17 oppure 23 in quanto, a parte la forza generata, la velocità la indicano uguale..poi dipende dal modello...oppure andare su servomotori ad anello chiuso...non so ancora

**Comment by phil-barrett** [comment]
I have been thinking about building a similar machine (Core XY, Laser).  My biggest concern is how much weight (mass) can a core XY belt driven mechanism move easily.  I would like to place a 40W optical power Laser Diode with power cable and air assist line.

As to motors - I will use Nema 23.  I do not trust Nema17 to have enough power.  

Stepper Drivers - To get the best speed, I want to run at the highest possible voltage.  The TB6600 is not a good choice because of voltage - the highest I would run them at is 36V.  The DM556 can run at 48V.  Also, most drivers called "TB6600" actually do not use the Toshiba TB6600 chip. 

I think closed loop would be good but perhaps "overkill" - i.e. more than you would need and a lot more expensive.

I would like to follow your build.  I am very interested in what you are doing.

Phil

**Reply by distebia** [reply]
ne ho costruita una da 700x500, mks dlc32 e driver drv8825, cinghie gates da 6mm, guide mgn9 su Y1 e Y2 e mgn9 sull'asse x (una sopra e una sotto per evitare oscillazioni del laser (ir-2w)...tutto con profili 2020 in alluminio...lo so che è esagerata una struttura core xy per un fibra da 2w ma la volevo fare per capirne le potenzialità...adesso ne vorrei fare una con profili 2040, teensy+t41u5xbb, dm556, laser co2, ancora sono a niente, solo qualche disegno in sketchup...la mia più grande paura è che, dopo aver completato la struttura, ho sempre dei limiti nell'incisione di foto, attualmente con teensy e co2 (cartesiano) subisco dei rallentamenti e scatti nel movimento e sono costretto a rallentare

**Reply by distebia** [reply]
circa il peso, la testa laser co2 dovrebbe pesare meno di un diodo, ma devo comunque portare un cavo x sensore induttivo (per homing dell'asse Z) e tubo dell'aria

**Reply by distebia** [reply]
> Ho pensato di costruire una macchina simile (Core XY, Laser). La mia preoccupazione più grande è quanto peso (massa) può spostare facilmente un meccanismo a cinghia Core XY. Vorrei posizionare un diodo laser ottico da 40 W con cavo di alimentazione e linea di assistenza ad aria.
> 
> Per quanto riguarda i motori, userò i Nema 23. Non credo che i Nema 17 abbiano abbastanza potenza.
> 
> Driver stepper: per ottenere la migliore velocità, voglio funzionare alla tensione più alta possibile. Il TB6600 non è una buona scelta a causa della tensione: la più alta a cui li farei funzionare è 36 V. Il DM556 può funzionare a 48 V. Inoltre, la maggior parte dei driver denominati "TB6600" in realtà non utilizza il chip Toshiba TB6600.
> 
> Penso che il circuito chiuso sarebbe una buona soluzione, ma forse "esagerato", ovvero più di quanto necessario e molto più costoso.
> 
> Vorrei seguire la tua build. Sono molto interessato a quello che stai facendo.
> 
> Filone

finalmente mi sono deciso...sono pronto per comprare un teensy 4.1 e una t41uxbb su tendie...collegamento ethernet, struttura core xy da 1000x1000 con nema 23 e dm556 alimentati a 48v. Grblhal con accelerazioni jerk...ecc...
Nella mia precedente struttura da 1300x1000 (cartesiana) utilizzo tb6600 (forse non originali) e impostavo $0=3...i tb6600 sono indicati come frequenza pulse di 20khz, contro i 200khz dei tb6600, quindi molto probabilmente i tb6600 mi limitavano nelle incisioni di immagini, infatti non riuscivo ad andare oltre i 20000mm/min...sono sulla strada giusta questa volta? hai dei consigli da darmi, sia lato hardware che software? Cosa dovrei affinare in grblhal (ed eventualmente compilarlo) per ottenere una velocità alta nelle incisioni di foto con un movimento fluido e non a scatti?

**Reply by phil-barrett** [reply]
If you were using an Arduino, then that is what limited the step rate to the driver. $0=3 is good. There are many factors that affect speed, I do not know if the TB6600 is the limiting one.

**Reply by distebia** [reply]
mi scuso per non essere stato chiaro...nella mia precedente costruzione (cartesiana 1300x1000) utilizzo un teensy su t41uxbb e 3 tb6600 alimentati a 36v, $398 anche aumentato a 1000, lb_cluster attivato e non attivato, $0=3...con questa configurazione non riesco a fare foto ad alta velocità, forse i tb6600 con 20khz non reggono il $0=3...

**Reply by phil-barrett** [reply]
Sorry!  $0=3 is 3 microSeconds pulse. That will allow approximately 250 kHz step rate.

Are you missing steps with the TB6600s? What is your acceleration on each axis - $12x?

**Reply by distebia** [reply]
con i tb6600 non perdo passi, ma vedo il movimento come impacciato e a scatti. se genera 250 khz allora funziona a meraviglia con i dm556, con i tb6600 non credo perchè funzionano a 20 khz e in qualche modo si creava un collo di bottiglia. Come accelerazione ho 2000 su X e 200 su Y (su Y posso mettere anche 2000 ma genera troppe oscillazioni del ponte Y), allego i valori $$

[setting.txt](https://github.com/user-attachments/files/18277767/setting.txt)

**Reply by phil-barrett** [reply]
It sounds like you are on the right track. The TB6600s that I have used often made grinding noises and were not that smooth even when driven at a low step rate. I would move to the DM556. I have several and they work well. I have had good luck with TMC5160 based drivers, also.

**Reply by distebia** [reply]
ok, almeno ho capito che il limite della mia cartesiana sono i tb6600, presto li sostituisco con dm556. prima ti ho allegato le impostazioni della cartesiana, qualcosa da cambiare oltre il $121? 
Per quanto riguarda la corexy in costruzione, teensy su tua scheda, driver dm556 a 48v e nema23, tutto su guide lineari.....grblhal con lb_cluster e accelerazioni jerk...che velocità potrei raggiungere nelle foto con toni di grigio?

**Reply by phil-barrett** [reply]
I don't know how fast it could go. Terje has seen a Teensy do better than 20,000 mm/min engraving speed. But, it depends on many things including motors/drivers, gearing, laser power.

**Reply by distebia** [reply]
ok, ti aggiorno, procedo con gli ordini...

**Reply by distebia** [reply]
> Ho pensato di costruire una macchina simile (Core XY, Laser). La mia preoccupazione più grande è quanto peso (massa) può spostare facilmente un meccanismo a cinghia Core XY. Vorrei posizionare un diodo laser ottico da 40 W con cavo di alimentazione e linea di assistenza ad aria.
> 
> Per quanto riguarda i motori, userò i Nema 23. Non credo che i Nema 17 abbiano abbastanza potenza.
> 
> Driver stepper: per ottenere la migliore velocità, voglio funzionare alla tensione più alta possibile. Il TB6600 non è una buona scelta a causa della tensione: la più alta a cui li farei funzionare è 36 V. Il DM556 può funzionare a 48 V. Inoltre, la maggior parte dei driver denominati "TB6600" in realtà non utilizza il chip Toshiba TB6600.
> 
> Penso che il circuito chiuso sarebbe una buona soluzione, ma forse "esagerato", ovvero più di quanto necessario e molto più costoso.
> 
> Vorrei seguire la tua build. Sono molto interessato a quello che stai facendo.
> 
> Filone

@phil-barrett Salve, le ho girato il link di SketchUp su messenger, magari qualche consiglio o critica?

**Reply by phil-barrett** [reply]
Facebook Messenger?  I do not see the link.

**Reply by distebia** [reply]
ti ho inviato il link su messenger/facebook...se non lo visualizzi dovresti controllare su richieste di messaggi o spam

**Reply by phil-barrett** [reply]
Definitely not seeing it.  Do you have the right phil barrett on facebook (there are several)?

**Reply by distebia** [reply]
![image](https://github.com/user-attachments/assets/d3d15549-2de6-4244-9246-b3cac2d186ca)
forse sbaglio persona?

**Reply by phil-barrett** [reply]
That is my picture.  I did a search for you in messenger and it found you but it forced me to create a PIN to "restore" my messenger contacts.  Once I did that, it showed your message with the link.  Not very user friendly.  sigh.

Anyway, thanks for the link will study it.  Good that you color coded the upper and lower belts.

**Reply by distebia** [reply]
grazie per aver aperto il link...non sono un bravo disegnatore e riesco a fare il minimo, ma mi basta per avere un'idea di base....il link ti porta sempre alla versione aggiornata. l'idea è quella di usare teensy, lasergrbl con jerk e lb_cluster, dm556 a 48v e nema 23...vediamo se la struttura core xy mi permette di avere alte velocità per il co2
